### ISSUE TITLE
fatal error has been detected by the Java Runtime Environment

### ISSUE THREAD

**Issue by komalsrathi** [bcbio/bcbio-nextgen#2453]
Hi,

I am facing another issue related to Java. I have tried this on 3 different servers, with different memory and core settings but all have the same problem. Here are the logs: 
[logs.txt](https://github.com/bcbio/bcbio-nextgen/files/2198051/logs.txt)

Here is my command:
```
/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/local/bin/bcbio_nextgen.py config.yaml -n 20
```

Config file:

```
details:
- algorithm:
    aligner: star
    variantcaller: gatk-haplotype
    # tools_off: gatk4
  analysis: RNA-seq
  description: 'Sample 1: CDL-634-13P'
  files:
  - /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/fastq/CDL-634-13P/CDL-634-13P_pe_1.fq.gz
  - /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/fastq/CDL-634-13P/CDL-634-13P_pe_2.fq.gz
  genome_build: GRCh37
  metadata:
    batch: CDL-634-13P
fc_name: CDL-634-13P-CDLS
upload:
  dir: /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/bcbio_out/final
resources:
  gatk:
    jvm_opts: ["-Xms2g", "-Xmx4g"]
```

Also, if there is no way this can be fixed then as an alternative how can I skip this step? My goal is to finish the pipeline and just get variant calls in a normal vcf format (I also don't care about the gvcf) from a single sample.

**Comment by chapmanb** [comment]
Komal;
Thanks for following up and for finding a fix for the genomicsdb native library loading. For this core dump error, can you replicate outside of bcbio by running the failing command? It'll be listed in `log/bcbio-nextgen-commands.log` when running.

If so, my first suggestion would be to try and determine the space of where it happens. Does it still core dump with using a single core and a small amount of memory? At what point does it start to trigger problems? This might give us some clues for debugging and working around. Thanks again for the help working through these issues.

**Comment by komalsrathi** [comment]
Hi,

Here is the log when I run it in stand alone mode:

```
# grep the failing command
[rathik@reslnrefo01 log]$ grep GenomicsDB bcbio-nextgen-commands.log
[2018-07-19T06:30Z] unset JAVA_HOME && export PATH=/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/bin:$PATH && gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpnSYdjf' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz

# run the command
[rathik@reslnrefo01 log]$ unset JAVA_HOME && export PATH=/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/bin:$PATH && gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpnSYdjf' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Using GATK jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpnSYdjf -jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/tmp/rathik
10:58:41.778 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
10:58:42.933 INFO  GenomicsDBImport - ------------------------------------------------------------
10:58:42.934 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.6.0
10:58:42.934 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
10:58:42.936 INFO  GenomicsDBImport - Executing as rathik@reslnrefo01.research.chop.edu on Linux v3.10.0-862.el7.x86_64 amd64
10:58:42.937 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_144-b01
10:58:42.937 INFO  GenomicsDBImport - Start Date/Time: July 19, 2018 10:58:41 AM EDT
10:58:42.937 INFO  GenomicsDBImport - ------------------------------------------------------------
10:58:42.937 INFO  GenomicsDBImport - ------------------------------------------------------------
10:58:42.938 INFO  GenomicsDBImport - HTSJDK Version: 2.16.0
10:58:42.938 INFO  GenomicsDBImport - Picard Version: 2.18.7
10:58:42.938 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
10:58:42.938 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
10:58:42.939 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
10:58:42.939 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
10:58:42.939 INFO  GenomicsDBImport - Deflater: IntelDeflater
10:58:42.939 INFO  GenomicsDBImport - Inflater: IntelInflater
10:58:42.939 INFO  GenomicsDBImport - GCS max retries/reopens: 20
10:58:42.939 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
10:58:42.939 INFO  GenomicsDBImport - Initializing engine
10:58:44.090 INFO  IntervalArgumentCollection - Processing 249250621 bp from intervals
10:58:44.123 INFO  GenomicsDBImport - Done initializing engine
Created workspace /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/CDL-164-04P-1_0_249250621_genomicsdb
10:58:44.723 INFO  GenomicsDBImport - Vid Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/vidmap.json
10:58:44.723 INFO  GenomicsDBImport - Callset Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/callset.json
10:58:44.723 INFO  GenomicsDBImport - Complete VCF Header will be written to CDL-164-04P-1_0_249250621_genomicsdb/vcfheader.vcf
10:58:44.723 INFO  GenomicsDBImport - Importing to array - CDL-164-04P-1_0_249250621_genomicsdb/genomicsdb_array
10:58:44.724 INFO  ProgressMeter - Starting traversal
10:58:44.724 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
10:58:45.326 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
Buffer resized from 22726bytes to 32529
Buffer resized from 32529bytes to 32693
Buffer resized from 32693bytes to 32738
Buffer resized from 32738bytes to 32741
Buffer resized from 32741bytes to 32756
Buffer resized from 32756bytes to 32768
Buffer resized from 32768bytes to 32769
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f1ab23b8fe4, pid=92028, tid=0x00007f1a9817e700
#
# JRE version: OpenJDK Runtime Environment (8.0_144-b01) (build 1.8.0_144-b01)
# Java VM: OpenJDK 64-Bit Server VM (25.144-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libc.so.6+0x153fe4]  __memmove_ssse3+0x2834
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/hs_err_pid92028.log
#
# If you would like to submit a bug report, please visit:
#   http://www.azulsystems.com/support/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

Here it is only using 1 core I am assuming.

**Comment by chapmanb** [comment]
Thanks for the additional debugging and outputs. It looks like there is some kind of issue with the Java installed with bcbio and your operating system. It's core dumping here talking with the instruction set. My best suggestion would be to try to work around this by using a system installed Java 8 if you have one on the machine. You can do this by setting `BCBIO_JAVA_HOME` prior to running bcbio:

https://bcbio-nextgen.readthedocs.io/en/latest/contents/installation.html?highlight=bcbio_java_home#automated

Fingers crossed using a different Java will resolve your problems.

**Comment by komalsrathi** [comment]
Just one confusion, by default i am using the system's java and BCBIO_JAVA_HOME is actually empty. Is that what you'd expect? 

```
[rathik@reslnrefo01 log]$ which java
/usr/bin/java
[rathik@reslnrefo01 log]$ echo $BCBIO_JAVA_HOME

```

**Comment by chapmanb** [comment]
Komal;
By default, bcbio uses an internally installed Java to try and avoid the many conflicts that can happen with different external installs (incorrect versions and other issues). In this case you want to circumvent that so can do:
```
export BCBIO_JAVA_HOME=/usr
```
and it should use your locally installed version. It would be useful to know if this helps at all with the segfaults. Thanks again for the help debugging.

**Comment by komalsrathi** [comment]
I tried setting the BCBIO_JAVA_HOME variable to internal java and it still has failed:

```
[rathik@reslnrefo01 log]$ export BCBIO_JAVA_HOME=/usr/bin/java

[rathik@reslnrefo01 log]$ echo $BCBIO_JAVA_HOME 
/usr/bin/java

[rathik@reslnrefo01 log]$ unset JAVA_HOME && export PATH=/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/bin:$PATH && gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpocpLmN' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Using GATK jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpocpLmN -jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/tmp/rathik
10:15:39.918 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
10:15:40.143 INFO  GenomicsDBImport - ------------------------------------------------------------
10:15:40.143 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.6.0
10:15:40.143 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
10:15:40.145 INFO  GenomicsDBImport - Executing as rathik@reslnrefo01.research.chop.edu on Linux v3.10.0-862.el7.x86_64 amd64
10:15:40.145 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_144-b01
10:15:40.145 INFO  GenomicsDBImport - Start Date/Time: July 20, 2018 10:15:39 AM EDT
10:15:40.145 INFO  GenomicsDBImport - ------------------------------------------------------------
10:15:40.145 INFO  GenomicsDBImport - ------------------------------------------------------------
10:15:40.146 INFO  GenomicsDBImport - HTSJDK Version: 2.16.0
10:15:40.146 INFO  GenomicsDBImport - Picard Version: 2.18.7
10:15:40.147 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
10:15:40.147 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
10:15:40.147 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
10:15:40.147 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
10:15:40.147 INFO  GenomicsDBImport - Deflater: IntelDeflater
10:15:40.147 INFO  GenomicsDBImport - Inflater: IntelInflater
10:15:40.147 INFO  GenomicsDBImport - GCS max retries/reopens: 20
10:15:40.147 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
10:15:40.148 INFO  GenomicsDBImport - Initializing engine
10:15:41.076 INFO  IntervalArgumentCollection - Processing 249250621 bp from intervals
10:15:41.083 INFO  GenomicsDBImport - Done initializing engine
Created workspace /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/CDL-164-04P-1_0_249250621_genomicsdb
10:15:41.502 INFO  GenomicsDBImport - Vid Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/vidmap.json
10:15:41.502 INFO  GenomicsDBImport - Callset Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/callset.json
10:15:41.503 INFO  GenomicsDBImport - Complete VCF Header will be written to CDL-164-04P-1_0_249250621_genomicsdb/vcfheader.vcf
10:15:41.503 INFO  GenomicsDBImport - Importing to array - CDL-164-04P-1_0_249250621_genomicsdb/genomicsdb_array
10:15:41.503 INFO  ProgressMeter - Starting traversal
10:15:41.503 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
10:15:41.896 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
Buffer resized from 22726bytes to 32529
Buffer resized from 32529bytes to 32693
Buffer resized from 32693bytes to 32738
Buffer resized from 32738bytes to 32741
Buffer resized from 32741bytes to 32756
Buffer resized from 32756bytes to 32768
Buffer resized from 32768bytes to 32769
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007efc8074f359, pid=143400, tid=0x00007efc803f6700
#
# JRE version: OpenJDK Runtime Environment (8.0_144-b01) (build 1.8.0_144-b01)
# Java VM: OpenJDK 64-Bit Server VM (25.144-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtiledbgenomicsdb5661446623350292251.so+0x155359]  BufferVariantCell::set_cell(void const*)+0x99
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/hs_err_pid143400.log
#
# If you would like to submit a bug report, please visit:
#   http://www.azulsystems.com/support/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

**Comment by chapmanb** [comment]
Komal;
Thanks for testing and for the report. Please try setting it as:
```
export BCBIO_JAVA_HOME=/usr
```
It's meant to be the equivalent to JAVA_HOME, not a reference to the `java` executable, so bcbio doesn't use it in your case. If it's working correctly you should see the java referenced in the command as your locally installed version rather than the bcbio installed one. The command should also not have:
```
unset JAVA_HOME && export PATH=/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/bin:$PATH
```
which is bcbio referencing it's installed java rather than the system one. Hope this works for you.

**Comment by komalsrathi** [comment]
Hi Brad,

Let me know if this is okay:

```
[rathik@reslnrefo01 log]$ echo $BCBIO_JAVA_HOME 
/usr

[rathik@reslnrefo01 log]$ gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpocpLmN' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Using GATK jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xms454m -Xmx3181m -XX:+UseSerialGC -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/joint/gatk-haplotype-joint/CDL-164-04P/1/bcbiotx/tmpocpLmN -jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/tmp/rathik
10:34:26.710 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
10:34:26.886 INFO  GenomicsDBImport - ------------------------------------------------------------
10:34:26.886 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.6.0
10:34:26.886 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
10:34:26.887 INFO  GenomicsDBImport - Executing as rathik@reslnrefo01.research.chop.edu on Linux v3.10.0-862.el7.x86_64 amd64
10:34:26.887 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_144-b01
10:34:26.887 INFO  GenomicsDBImport - Start Date/Time: July 20, 2018 10:34:26 AM EDT
10:34:26.887 INFO  GenomicsDBImport - ------------------------------------------------------------
10:34:26.887 INFO  GenomicsDBImport - ------------------------------------------------------------
10:34:26.888 INFO  GenomicsDBImport - HTSJDK Version: 2.16.0
10:34:26.888 INFO  GenomicsDBImport - Picard Version: 2.18.7
10:34:26.888 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
10:34:26.888 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
10:34:26.888 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
10:34:26.888 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
10:34:26.888 INFO  GenomicsDBImport - Deflater: IntelDeflater
10:34:26.888 INFO  GenomicsDBImport - Inflater: IntelInflater
10:34:26.888 INFO  GenomicsDBImport - GCS max retries/reopens: 20
10:34:26.888 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
10:34:26.888 INFO  GenomicsDBImport - Initializing engine
10:34:27.442 INFO  IntervalArgumentCollection - Processing 249250621 bp from intervals
10:34:27.446 INFO  GenomicsDBImport - Done initializing engine
Created workspace /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/CDL-164-04P-1_0_249250621_genomicsdb
10:34:27.778 INFO  GenomicsDBImport - Vid Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/vidmap.json
10:34:27.778 INFO  GenomicsDBImport - Callset Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/callset.json
10:34:27.778 INFO  GenomicsDBImport - Complete VCF Header will be written to CDL-164-04P-1_0_249250621_genomicsdb/vcfheader.vcf
10:34:27.778 INFO  GenomicsDBImport - Importing to array - CDL-164-04P-1_0_249250621_genomicsdb/genomicsdb_array
10:34:27.778 INFO  ProgressMeter - Starting traversal
10:34:27.779 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
10:34:28.049 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
Buffer resized from 22726bytes to 32529
Buffer resized from 32529bytes to 32693
Buffer resized from 32693bytes to 32738
Buffer resized from 32738bytes to 32741
Buffer resized from 32741bytes to 32756
Buffer resized from 32756bytes to 32768
Buffer resized from 32768bytes to 32769
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f32f0f21359, pid=145690, tid=0x00007f335c1d7700
#
# JRE version: OpenJDK Runtime Environment (8.0_144-b01) (build 1.8.0_144-b01)
# Java VM: OpenJDK 64-Bit Server VM (25.144-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtiledbgenomicsdb2489195489540231512.so+0x155359]  BufferVariantCell::set_cell(void const*)+0x99
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/hs_err_pid145690.log
#
# If you would like to submit a bug report, please visit:
#   http://www.azulsystems.com/support/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

**Comment by chapmanb** [comment]
Komal;
Thanks for this. The `BCBIO_JAVA_HOME` only works when running from in bcbio, not using the `gatk` script separately (that's developed by the Broad team, not us). So if you want to test with this, which is fine, you only need to remove the bcbio specific path manipulation (which you did) and then ensure the java you want is first in your PATH (which I don't think is the case here since it's still using the bcbio one). So I'd suggest:
```
export PATH=/usr/bin:$PATH
gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
```
Thanks again for your patience and hope this uses your local java and helps give some insight.

**Comment by komalsrathi** [comment]
Hi,

Method 1: export BCBIO_JAVA_HOME, export PATH and run gatk. Also tried without the first line of exporting BCBIO_JAVA_HOME - gives same error:

```
[rathik@reslnrefo01 log]$ export BCBIO_JAVA_HOME=/usr
[rathik@reslnrefo01 log]$ export PATH=/usr/bin:$PATH
[rathik@reslnrefo01 log]$ gatk --java-options '-Xms454m -Xmx3181m -XX:+UseSerialGC' GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Using GATK jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xms454m -Xmx3181m -XX:+UseSerialGC -jar /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar GenomicsDBImport --reader-threads 1 --genomicsdb-workspace-path CDL-164-04P-1_0_249250621_genomicsdb -L 1:1-249250621 --variant /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/variation/rnaseq/gatk-haplotype/Sample_1__CDL-164-04P-gatk-haplotype-annotated-rnaedit-annotated-gemini.vcf.gz
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/mnt/isilon/cbmi/variome/tmp/rathik
11:49:24.784 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/mnt/isilon/cbmi/variome/bin/bcbio-nextgen/bcbio/anaconda/share/gatk4-4.0.6.0-0/gatk-package-4.0.6.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
11:49:25.130 INFO  GenomicsDBImport - ------------------------------------------------------------
11:49:25.130 INFO  GenomicsDBImport - The Genome Analysis Toolkit (GATK) v4.0.6.0
11:49:25.131 INFO  GenomicsDBImport - For support and documentation go to https://software.broadinstitute.org/gatk/
11:49:25.134 INFO  GenomicsDBImport - Executing as rathik@reslnrefo01.research.chop.edu on Linux v3.10.0-862.el7.x86_64 amd64
11:49:25.134 INFO  GenomicsDBImport - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_171-b10
11:49:25.134 INFO  GenomicsDBImport - Start Date/Time: July 20, 2018 11:49:24 AM EDT
11:49:25.134 INFO  GenomicsDBImport - ------------------------------------------------------------
11:49:25.134 INFO  GenomicsDBImport - ------------------------------------------------------------
11:49:25.135 INFO  GenomicsDBImport - HTSJDK Version: 2.16.0
11:49:25.135 INFO  GenomicsDBImport - Picard Version: 2.18.7
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.COMPRESSION_LEVEL : 2
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
11:49:25.136 INFO  GenomicsDBImport - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
11:49:25.136 INFO  GenomicsDBImport - Deflater: IntelDeflater
11:49:25.136 INFO  GenomicsDBImport - Inflater: IntelInflater
11:49:25.136 INFO  GenomicsDBImport - GCS max retries/reopens: 20
11:49:25.136 INFO  GenomicsDBImport - Using google-cloud-java patch 6d11bef1c81f885c26b2b56c8616b7a705171e4f from https://github.com/droazen/google-cloud-java/tree/dr_all_nio_fixes
11:49:25.137 INFO  GenomicsDBImport - Initializing engine
11:49:25.926 INFO  IntervalArgumentCollection - Processing 249250621 bp from intervals
11:49:25.931 INFO  GenomicsDBImport - Done initializing engine
Created workspace /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/CDL-164-04P-1_0_249250621_genomicsdb
11:49:26.339 INFO  GenomicsDBImport - Vid Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/vidmap.json
11:49:26.339 INFO  GenomicsDBImport - Callset Map JSON file will be written to CDL-164-04P-1_0_249250621_genomicsdb/callset.json
11:49:26.339 INFO  GenomicsDBImport - Complete VCF Header will be written to CDL-164-04P-1_0_249250621_genomicsdb/vcfheader.vcf
11:49:26.339 INFO  GenomicsDBImport - Importing to array - CDL-164-04P-1_0_249250621_genomicsdb/genomicsdb_array
11:49:26.339 INFO  ProgressMeter - Starting traversal
11:49:26.340 INFO  ProgressMeter -        Current Locus  Elapsed Minutes     Batches Processed   Batches/Minute
11:49:26.697 INFO  GenomicsDBImport - Importing batch 1 with 1 samples
Buffer resized from 22726bytes to 32529
Buffer resized from 32529bytes to 32693
Buffer resized from 32693bytes to 32738
Buffer resized from 32738bytes to 32741
Buffer resized from 32741bytes to 32756
Buffer resized from 32756bytes to 32768
Buffer resized from 32768bytes to 32769
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f9935bac359, pid=8320, tid=0x00007f99881bf700
#
# JRE version: OpenJDK Runtime Environment (8.0_171-b10) (build 1.8.0_171-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.171-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libtiledbgenomicsdb1661720680664773125.so+0x155359]  BufferVariantCell::set_cell(void const*)+0x99
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /mnt/isilon/cbmi/variome/rathik/mendelian_rnaseq/gatk_output/CDL-164-04P/log/hs_err_pid8320.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

Method 2: Export BCBIO_JAVA_HOME and run from within bcbio. Logs are attached after the command.

```
[rathik@reslnrefo01 CDL-164-04P]$ export BCBIO_JAVA_HOME=/usr
[rathik@reslnrefo01 CDL-164-04P]$ echo $BCBIO_JAVA_HOME 
/usr
[rathik@reslnrefo01 CDL-164-04P]$ /mnt/isilon/cbmi/variome/bin/bcbio-nextgen/local/bin/bcbio_nextgen.py config.yaml -n 1
```
[logs.txt](https://github.com/bcbio/bcbio-nextgen/files/2214565/logs.txt)

**Comment by chapmanb** [comment]
Komal;
Thanks much for testing. It looks from the core dump OpenJDK version that both methods used your locally installed java and are still failing in the same way. We can rule out a version specific issue; thank you.

Practically, I'm not sure how to debug this more from my side and think the next step is to report the issue on the GATK issue tracker (https://github.com/broadinstitute/gatk/issues) to see if folks there can help further. Please copy us in and we can help as need. They might suggest using/testing Oracle Java if you have that installed, or have other ideas for what is wrong and what to try.

Thanks again for your patience and sorry to not have more ideas about what is happening.

**Comment by komalsrathi** [comment]
Ok sure - will do. Did you want me to close this here?

**Comment by chapmanb** [comment]
Komal;
Thanks, if you can link to the GATK issue from here we can leave it open as a reminder to be sure it's resolved. There may be things we can adjust from our side, hopefully the GATK team will be able to provide some clues.

**Comment by komalsrathi** [comment]
GATK issue: https://github.com/broadinstitute/gatk/issues/5045

**Comment by komalsrathi** [comment]
@chapmanb whenever you get a chance, can you take a look at the gatk issue thread? https://github.com/broadinstitute/gatk/issues/5045#issuecomment-407476343

**Comment by chapmanb** [comment]
Komal;
Thanks again for this. As discussed in that thread, we pushed a fix to move the annotations after joint calling to avoid the problem. If you update to the latest development version:
```
bcbio_nextgen.py upgrade -u development
```
and re-run the variant calling part of your analysis (by removing the `variation` directory in your work folder) it should hopefully work by running annotations after this step. 

Thanks again for all the work debugging this. Please reopen if you're still hitting issues.

**Comment by komalsrathi** [comment]
Hi @chapmanb Thanks for the help. The run finished successfully! Just one last question - can you tell me which of these is the final variant file that I should be looking at? Here are the files generated:

```
# annotated vcf files
[rathik@reslnrefo01 CDL-164-04P]$ find . -name "*annotated*.vcf.gz"

./joint/gatk-haplotype-joint/CDL-164-04P/CDL-164-04P-joint-annotated-rnaedit-filter.vcf.gz
./joint/gatk-haplotype-joint/CDL-164-04P/CDL-164-04P-joint-annotated-rnaedit.vcf.gz
./2018-07-25_CDL-164-04P-CDLS/CDL-164-04P-gatk-haplotype-annotated.vcf.gz
```

**Comment by chapmanb** [comment]
Komal;
If the workflow finished successfully, the final variant call files should get moved into the `final` directory so you don't have to dig them out of the work directory. Of the ones you listed in work directory, `./joint/gatk-haplotype-joint/CDL-164-04P/CDL-164-04P-joint-annotated-rnaedit-filter.vcf.gz` if the final VCF. The other is a gVCF input and intermediate.

Thank you again for all the work debugging and glad this is now working for you.
